### PR TITLE
Fix end to end benchmark with rpc devices

### PR DESCRIPTION
**Pull request by tkonolige**
I forgot to convert remote devices to local devices in the `benchmark` rpc call. This PR fixes the issue and also adds a test to catch it. I've also improved the error message for the failure.

### CODE REVIEW

**Comment by tkonolige** [comment]
@areusch or @junrushao1994 Can you review?

File: tests/python/relay/test_backend_graph_executor.py [hunk 1]
```diff
@@ -361,21 +361,21 @@ def test_benchmark_end_to_end(dev, target):
...
 
-@tvm.testing.requires_llvm
+@tvm.testing.requires_cuda
 def test_benchmark_end_to_end_rpc():
     server = rpc.Server("127.0.0.1")
     remote = rpc.connect(server.host, server.port)
 
     mod, params = mlp.get_workload(1)
-    lib = relay.build(mod, target="llvm", params=params)
+    lib = relay.build(mod, target="cuda", params=params)
```
Comment (areusch) [inline tests/python/relay/test_backend_graph_executor.py:370]: why this change?

Comment (tkonolige) [reply tests/python/relay/test_backend_graph_executor.py:370]: The check that prevents you from copying from local to remote only allows copying from local cpu to remote cpu, so this test was not having any issues even though its behavior was wrong. (https://github.com/apache/tvm/pull/9175/files#diff-c2d3e12882a0e96941e57203f5f4c40eb7d9cf2e913d991fe8307f0930c2bb82R234-R238).

**Review by junrushao1994 — APPROVED** [review]
LGTM

File: tests/python/relay/test_backend_graph_executor.py (hunk 1 above)
Comment (areusch) [reply tests/python/relay/test_backend_graph_executor.py:370]: ah gotcha.

**Review by areusch — APPROVED** [review]
